### PR TITLE
add backupstoragelocation to velero pkg

### PR DESCRIPTION
**Pull request by trewest**
Add backupstoragelocation struct to velero pkg

### CODE REVIEW

File: pkg/velero/backupstoragelocation.go [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+	veleroClient "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// BackupStorageLocationBuilder provides a struct for backupstoragelocation
+// object from the cluster and a backupstoragelocation definition.
+type BackupStorageLocationBuilder struct {
+	// Backup definition, used to create the backupstoragelocation object.
```
Comment (achuzhoy) [inline pkg/velero/backupstoragelocation.go:21]: BackupStorageLocation definiton

File: pkg/velero/backupstoragelocation.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+// NewBackupStorageLocationBuilder creates a new instance of BackupStorageLocationBuilder.
+func NewBackupStorageLocationBuilder(
+	apiClient *clients.Settings,
+	name string,
+	namespace string,
+	provider string,
+	objectStorage velerov1.ObjectStorageLocation) *BackupStorageLocationBuilder {
+	glog.V(100).Infof(
+		"Initializing new backup structure with the following params: "+
```
Comment (achuzhoy) [inline pkg/velero/backupstoragelocation.go:39]:
```suggestion
		"Initializing new backupStorageLocation structure with the following params: "+
```

File: pkg/velero/backupstoragelocationlist.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ListBackupStorageLocationBuilder returns backupstoragelocation inventory in the given namespace.
```
Comment (achuzhoy) [inline pkg/velero/backupstoragelocationlist.go:12]: Can we have it in the same file with backupstoragelocation?

Comment (trewest) [reply pkg/velero/backupstoragelocationlist.go:12]: We put list functions into a separate file for organization purposes

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**